crates/oxc_minifier/src/ast_passes/peephole_minimize_conditions.rs

```diff
@@ -457,6 +458,37 @@ impl<'a> PeepholeMinimizeConditions {
     }
 }
 
+// returns `true` if the current node is in a context in which the return
+// value type is coerced to boolean.
+// For example `if (condition)` and `return condition`
+// inside the `if` stmt, `condition` is coerced to a boolean
+// whereas inside the return, it is not
+fn is_in_boolean_context(ctx: &mut TraverseCtx<'_>) -> bool {
+    for ancestor in ctx.ancestors() {
+        match ancestor {
+            Ancestor::IfStatementTest(_)
+            | Ancestor::WhileStatementTest(_)
+            | Ancestor::DoWhileStatementTest(_)
+            | Ancestor::ExpressionStatementExpression(_) => return true,
+            Ancestor::CallExpressionArguments(_)
+            | Ancestor::BindingRestElementArgument(_)
+            | Ancestor::JSXSpreadAttributeArgument(_)
+            | Ancestor::NewExpressionArguments(_)
+            | Ancestor::ObjectPropertyKey(_)
+            | Ancestor::ObjectPropertyValue(_)
+            | Ancestor::ReturnStatementArgument(_)
+            | Ancestor::ThrowStatementArgument(_)
+            | Ancestor::YieldExpressionArgument(_)
+            | Ancestor::VariableDeclaratorInit(_) => return false,
```

Comment:
not convinced this covers everything. but the test 262 suite passes and these are the only ones i can think of 😅